### ISSUE TITLE
☂️  Recipe Tool

### ISSUE THREAD

**Issue by autumnfjeld** [Automattic/wp-calypso#62683]
# Description

Part of Verticals in Onboarding bxlJb-p2
See PT: Verticalized pattern assembly to create page layouts: pbxlJb-XO-p2

Following the initial work in  #61788, we are ready to make this a user friendly tool with significant improvement in functionality. See MC tool  /verticals/#/recipes/

<img width="797" alt="image" src="https://user-images.githubusercontent.com/1525580/163349720-c9145b05-f079-4662-8db5-980aa5a9f75b.png">

# Example user flow
* User clicks Create Recipe
* User selects a theme 
* User selects a few patterns
  *  patterns appear in the preview window
* User selects a vertical
  * images in the patterns are replaced with images for the selected vertical 
* User saves recipe 
* User clicks "Generate mShots" button (or goes to mShots tab)
  * mShots preview is shown  

# Acceptance Criteria 

## Usability

- [ ] Add instructions to the tool. Maybe add some helper text below each input field.
  - [ ] Indicate how to find pattern names (e.g. open up dotcompatterns or boilerplate to find pattern name) 
- [x] Update theme list to only allow [themes that "work out of the box" ](https://github.com/Automattic/wp-calypso/issues/62724#issuecomment-1098691034) (e.g. not Geologist as it consists of query block).
   * D78841-code
- [x] Update theme list to include [Meraki](https://github.com/Automattic/themes/tree/trunk/meraki)  and the [other WIP themes](xjXEAnOQeTReSqfVmnTqlH-fi-1%3A54) as they become availble cc: @that-mike-bal 
   * D78841-code

## Vertical selection

- [x] Enable the vertical dropdown. Data is pulled from new verticals API.
   - [x] D78908-code 

## Image injection

- [x] When user selects vertical, pass the vertical id to the preview URL so that it can reload with images injected.
   - [x] D78267-code
- [x] First image injection can be totally random images tagged to that vertical (ignoring dimensions/semantic tagging etc)
    - [x] D78994-code   

## Saving recipes

- #63445
- [x] Add Save functionality. Maybe need a P2 for this.
   - [x] P2 for storing recipes: pbxlJb-1DZ-p2
   - [x] Implement list, create, get recipe endpoints: D79776-code
   - [x] Implement list and create recipe UI: D79777-code
- [x] Show list of existing recipes on  /verticals/#/recipes/
   - [x] Implement list, create, get recipe endpoints: D79776-code
   - [x] Implement list and create recipe UI: D79777-code
- [ ] Approved by Autumn 

## Viewing and updating recipes
- [x] Allow viewing existing recipe (by clicking on the recipe)
- [x] Allow updating existing recipe
- [x] Allow deleting a recipe

## mShots (nice to have, not required)

- [ ] Show the mShots / allow user to view mShots. The layout of iframe & mShots TBD.

# Ideas for Subsequent Versions

* Show pattern suggestions as a dropdown list.
* User can rearrange order of block patterns.

# Testing
- 

# Related
Project Thread:  pbxlJb-XO-p2
MC Tool for Generated Vertical Designs #61788

**Comment by that-mike-bal** [comment]
Hey there, I just want to make a quick note on the vertical selection UI. 

Based on upcoming flows being goal focused, we likely won't need to map specific recipes to specific verticals. Each recipe *can* work across a wide variety of verticals pending on what type of site it is or what the user is trying to achieve with that site. 

For the MVP: Since @iamtakashi created different recipes for each vertical I think we're ok assigning one vertical per recipe but in the future, if we needed to keep tagging them with verticals, we would need to be able to select multiple verticals. Again, I don't think it will make sense to tag recipes with verticals based on the direction we're going but I just wanted to put that out there for consideration.

**Comment by autumnfjeld** [comment]
@that-mike-bal @Automattic/ganon  

> one vertical per recipe

We don't want to assign a vertical to a recipe. Let's stick with our original plan, as described in the MVP pbxlJb-1tU-p2,  of showing three types of sites per vertical. 

![image](https://user-images.githubusercontent.com/5665959/164133000-287de049-5fbd-4a91-9a5e-dfa1d5c522da.png)

We need to think of "growing verticals" as a continuous process, we should be adding new image verticals every day - in theory we _could_ easily have content (images) for hundreds of verticals by the time we launch to customers -- if we have someone dedicated to image uploads.

If we have 3 blog recipes, 3 portfolio recipes, 3 business service recipes we could  randomize which is served.   Each user would see a blog, a portfolio, & a biz service, but we can randomly serve these from the selection of blog, portfolio, & biz services.    If we did this, then I see a need to categorize the recipes by site type, which we could easily do this by naming them Blog1, Blog2, Blog3.

**Comment by that-mike-bal** [comment]
> If we have 3 blog recipes, 3 portfolio recipes, 3 business service recipes we could randomize which is served. Each user would see a blog, a portfolio, & a biz service, but we can randomly serve these from the selection of blog, portfolio, & biz services. If we did this, then I see a need to categorize the recipes by site type, which we could easily do this by naming them Blog1, Blog2, Blog3.

We can do this with the recipes/layouts @iamtakashi put together for us as soon as the recipe tool is ready to go.

**Comment by fushar** [comment]
@autumnfjeld for this one:

> Add instructions to the tool. Maybe add some helper text below each input field.

I think now it makes more sense to write up a FG page instead, similar to the image curator tool. The feature set has become more complex. What do you think?

**Comment by miksansegundo** [comment]
@fushar I agree. An FG page with screenshots could be a better resource for learning how to use the tool and easier to maintain. I think that on the header of the Recipe tool we can add a short description and the link to the FG. Would that work?

**Comment by fushar** [comment]
That works as discussed in our internal Slack, thanks! Now we can close this issue 😄
